.travis.yml

```diff
@@ -54,11 +54,11 @@ jobs:
       install:
         - rm -rf /opt/cita-run/test-chain
         - cd $TRAVIS_BUILD_DIR/target/install
-        - ./env.sh ./scripts/create_cita_config.py create
+        - ./bin/cita create
```

Comment:
maybe use bebop?

Reply:
Might not, 

The official branch use `env.sh`, 

And, every command needs `./env.sh` or `daemon.sh` converts to use `bebop` or `bebop daemon` automatically in my branch XD, 

`bebop` keyword only required when scripts running directly without docker.

Reply:
oops, the No.55 line, seems the commands are in docker?

Reply:
> bebop keyword only required when scripts running directly without docker.

this is the situation

Reply:
I notice that the failed job is still using the old config.

Would you please to rebase the latest branch and squash some commits.

Reply:
I'm afraid I couldn't do it well, I'm a bad __rebaser__, last time rebasing exchanging my 3 force push....XD, but I will try it again anyway.

Reply:
You are right, No need to use `bebop`

Reply:
> I'm afraid I couldn't do it well, I'm a bad **rebaser**, last time rebasing exchanged my 3 force push....XD, but I will try it again anyway.

remeber to use `rebase -i develop`. you can try to keep your commit clean or maybe we can use the squash merging.

Reply:
Hey @kaikai1024 , could you checkout my [testing][1] branch, if I did the right thing, should I do the same to my develop branch, and __force push_...again?

[1]: https://github.com/clearloop/cita/commits/testing

Reply:
If the concrete commands I should use are:

```
git rebase -i Head~9
# ~> squash the commits
# ~> do some commit describes
# ~> and then, my branch became clean.
git push -f
```

Reply:
I'm not sure about the remote info
1. check it using  `git remote -v`(assume that cryptape/cita is origin)
2. `git fetch origin/develop`
3. `git rebase -i origin/develop`

Reply:
Could i clone your repo and creae a new branch on my repo, you can check the commits.

Reply:
We can do it later. let's focus on the travisCI.

Reply:
Okay, but it seems I just solve the squash problem, [here][1] it is.

All I confused is if I have to __force push__.

The ci's problem is all about the JSONRPC, no response, I've written the condition about `--net=host` when it's not running on macos, and I'm thinking about why it happens now.

[1]: https://github.com/clearloop/cita/commits/testing

Reply:
ok.

I will try to fix the travis.
Actually, you just run the travis on your own repo, it's much faster

Reply:
ok.
I am trying to fix it on my branch:
https://github.com/kaikai1024/cita/tree/fix_travis_cita

* rebase the later develop
* fixup other commits
* reword the commit

Reply:
> Hey @kaikai1024 , could you checkout my testing branch?

That's right

Reply:
CI's scripts copies the `target/install-cache` into `target/install` after deleting it:
```
 before_install:
    - cd $TRAVIS_BUILD_DIR
    - rm -rf target/install
    - cp -rv target/release-cache target/install
```

The error might about the exec grant of the copied `cita-env`, 

Here are the logs of my latest build: 
```
$ ./bin/cita start test-chain/0
cita-env exists
error: exec: "/home/travis/build/clearloop/cita/target/install/bin/cita-env": stat /home/travis/build/clearloop/cita/target/install/bin/cita-env: no such file or directory
```

I wrote exists' tests, The file is exist, but it isn't executable, so I find this same [question][1].

I'm quite confused now, `cita` has the executable grant, why `cita-env` and `cita-config` not? even I `chmod +x` them in `release.sh`.

[1]: https://askubuntu.com/questions/133389/no-such-file-or-directory-but-the-file-exists

Reply:
Maybe something with the cache of travis.
Wait a minute, I'll try it.

Reply:
I have the same problem on my pc.